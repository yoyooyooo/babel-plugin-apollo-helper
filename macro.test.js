const pluginTester = require('babel-plugin-tester');
const path = require('path');
const plugin = require('babel-plugin-macros');

pluginTester.default({
  pluginName: 'apollo-helper',
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename, parserOpts: { plugins: ['jsx'] } },
  tests: {
    '1 query': `
      import autoInjectGql from "./macro";
      import { graphql } from '@apollo/react-hoc';
      import compose from 'lodash/flowRight';

      const App = props => {
        return (
          <div>
            <span>1</span>
          </div>
        );
      };

      export const query = gql\`
        query queryDemo {
          a {
            b
          }
        }
      \`;
      export default autoInjectGql(App);
    `,
    '2 query': `
      import autoInjectGql from "./macro";
      import { graphql } from '@apollo/react-hoc';
      import compose from 'lodash/flowRight';

      const App = props => {
        return (
          <div>
            <span>1</span>
          </div>
        );
      };

      export const query1 = gql\`
        query queryDemo1 {
          a {
            b
          }
        }
      \`;
      export const query2 = gql\`
        query queryDemo2 {
          c {
            d
          }
        }
      \`;
      export default autoInjectGql(App);
    `
  }
});
