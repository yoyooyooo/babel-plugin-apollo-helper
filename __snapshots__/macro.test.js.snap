// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apollo-helper 1 query: 1 query 1`] = `

import autoInjectGql from "./macro";
import { graphql } from '@apollo/react-hoc';
import compose from 'lodash/flowRight';

const App = props => {
  return (
    <div>
      <span>1</span>
    </div>
  );
};

export const query = gql\`
  query queryDemo {
    a {
      b
    }
  }
\`;
export default autoInjectGql(App);

      ↓ ↓ ↓ ↓ ↓ ↓

import { graphql } from '@apollo/react-hoc';
import compose from 'lodash/flowRight';

const App = props => {
  return (
    <div>
      <span>1</span>
    </div>
  );
};

export default graphql(
  gql\`
    query queryDemo {
      a {
        b
      }
    }
  \`,
  {
    name: 'query'
  }
)(App);


`;

exports[`apollo-helper 2 query: 2 query 1`] = `

import autoInjectGql from "./macro";
import { graphql } from '@apollo/react-hoc';
import compose from 'lodash/flowRight';

const App = props => {
  return (
    <div>
      <span>1</span>
    </div>
  );
};

export const query1 = gql\`
  query queryDemo1 {
    a {
      b
    }
  }
\`;
export const query2 = gql\`
  query queryDemo2 {
    c {
      d
    }
  }
\`;
export default autoInjectGql(App);

      ↓ ↓ ↓ ↓ ↓ ↓

import { graphql } from '@apollo/react-hoc';
import compose from 'lodash/flowRight';

const App = props => {
  return (
    <div>
      <span>1</span>
    </div>
  );
};

export default compose(
  graphql(
    gql\`
      query queryDemo1 {
        a {
          b
        }
      }
    \`,
    {
      name: 'query1'
    }
  ),
  graphql(
    gql\`
      query queryDemo2 {
        c {
          d
        }
      }
    \`,
    {
      name: 'query2'
    }
  )
)(App);


`;
